### ISSUE TITLE
Kernel for ctable operation

### ISSUE THREAD

**Issue by pdamme** [daphne-eu/daphne#390]
This task is to implement a kernel for DaphneIR's `CTableOp`. The contingency table of two *(n x 1)* matrices *lhs* and *rhs* is a matrix *res* of shape (*max(lhs) + 1*, *max(rhs) + 1*), and *res[i, j] = |{ k | A[k] = i and B[k] = j, 0 ≤ k ≤ n-1 }|*.

Example:
```
lhs: 2
     4
     5
     4

rhs: 0
     3
     1
     3

res: 0 0 0 0
     0 0 0 0
     1 0 0 0
     0 0 0 0
     0 0 0 2
     0 1 0 0
```

Note that this operation is inspired by SystemDS's `table()` built-in function (see https://apache.github.io/systemds/site/dml-language-reference). However, its definition is slightly different here, since we start counting rows/columns at zero (not at one).

We currently need this kernel to make the PCA DaphneDSL script in `scripts/algorithms/pca.daph` run.

**Comment by akroviakov** [comment]
Yes sorry, my mistake, I've misinterpreted the brackets in the definition.
